.github/workflows/build_for_pypi.yml

```diff
@@ -1,46 +1,90 @@
+---
 name: Build and Optionally Publish to PyPi
 
 on:
-  workflow_call:
-    inputs:
-      publish:
-        type: boolean
-        default: false
-        description: "Publish to PyPi"
+  pull_request:
+    branches:
+      - main
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
 
 jobs:
-  build_for_pypi:
+  build_dist_for_pypi:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.22.0
+        env:
+          CIBW_SKIP: pp*
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: cibw-wheels-${{ matrix.os }}
```

Comment:
With the matrix getting more factors in (https://github.com/codecov/codecov-cli/pull/604/files#r1930741461), don't forget to adjust this name too.